CMakeLists.txt

```diff
@@ -4,7 +4,13 @@ project(stack_task VERSION 0.1.0)
 include(CTest)
 enable_testing()
 
-add_executable(stack_task main.c)
+ set(SOURCE_FILES 
+  src/stack/stack.c
+  src/balanced_task/balanced_task.c
+  src/main.c
+  )
+
+ add_executable(stack_task ${SOURCE_FILES})
```

Comment:
It seems to be better if you use a such approach to sourcing files

add_executable(${PROJECT_NAME})

target_sources

target_precompile_headers